Makefile

```diff
@@ -122,62 +123,75 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-ifneq (, $(SKIP_TESTS))
-	@echo Skipping tests. Unset SKIP_TESTS to actually run them.
-else
-	# Just build the test and set write flag on created folder, so that we can clean it up
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(GOFLAGS) ./... -coverprofile cover.out -c
-	chmod +w $(LOCALBIN)/k8s/$(ENVTEST_K8S_VERSION)*
-	# Then run the test
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(GOFLAGS) ./... -coverprofile cover.out
-endif
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $$(go list ./... | grep -v /e2e) -coverprofile cover.out
+
+# Utilize Kind or modify the e2e tests to load the image locally, enabling compatibility with other vendors.
+.PHONY: test-e2e  # Run the e2e tests against a Kind k8s instance that is spun up.
+test-e2e:
+	go test ./test/e2e/ -v -ginkgo.v
+
+.PHONY: lint
+lint: golangci-lint ## Run golangci-lint linter & yamllint
+	$(GOLANGCI_LINT) run
+
+.PHONY: lint-fix
+lint-fix: golangci-lint ## Run golangci-lint linter and perform fixes
+	$(GOLANGCI_LINT) run --fix
 
 ##@ Build
 
 .PHONY: build
-build: generate fmt vet ## Build manager binary.
-	CGO_ENABLED=0 GOOS=linux go build $(GOFLAGS) -mod=mod -o bin/manager main.go
+build: manifests generate fmt vet ## Build manager binary.
+	CGO_ENABLED=0 GOOS=linux go build $(GOFLAGS) -o bin/manager main.go
 
 .PHONY: run
 run: manifests generate fmt vet ## Run a controller from your host.
 	go run ./main.go
 
+# If you wish to build the manager image targeting other platforms you can use the --platform flag.
+# (i.e. docker build --platform linux/arm64). However, you must enable docker buildKit for it.
+# More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
-docker-build: test ## Build docker image with the manager.
-	docker build \
+docker-build: ## Build docker image with the manager.
```

Comment:
@pmores check this build target. Earlier docker-build always called `test` target

Reply:
On a safer side, do you recommend keeping SKIP_TESTS as-is?

Reply:
@bpradipt where does the `test` dep come from ? SDK ?

Reply:
yes

Reply:
I'm wrong. The test dep was infact added by me ;) 
```
c579d7f5 (Pradipta Banerjee      2021-09-08 03:39:56 -0400 146) docker-build: test ## Build docker image with the manager.
39574e79 (John Allberg           2023-03-09 21:10:45 +0100 147)         docker build \
39574e79 (John Allberg           2023-03-09 21:10:45 +0100 148)                 -t ${IMG} \
```

Reply:
Cool, let's drop it and match the SDK then :smile: